pkg/controllers/kubernetes/secret.go

```diff
@@ -0,0 +1,53 @@
+package kubernetes
+
+import (
+	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/endpoints"
+	logging "github.com/tektoncd/dashboard/pkg/logging"
+	v1 "k8s.io/api/core/v1"
+	k8sinformer "k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+func NewSecretController(sharedK8sInformerFactory k8sinformer.SharedInformerFactory) {
+	logging.Log.Debug("In NewSecretController")
+	k8sInformer := sharedK8sInformerFactory.Core().V1().Secrets()
+	k8sInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    secretCreated,
+		UpdateFunc: secretUpdated,
+		DeleteFunc: secretDeleted,
+	})
+}
+
+func secretCreated(obj interface{}) {
+	logging.Log.Debug("Secret Controller Create")
+	data := broadcaster.SocketData{
+		MessageType: broadcaster.SecretCreated,
+		Payload:     obj,
+	}
+
+	endpoints.ResourcesChannel <- data
+}
+
+func secretUpdated(oldObj, newObj interface{}) {
+	oldSecret, newSecret := oldObj.(*v1.Secret), newObj.(*v1.Secret)
+	// If resourceVersion differs between old and new, an actual update event was observed
+	if oldSecret.ResourceVersion != newSecret.ResourceVersion {
+		logging.Log.Debug("Secret Controller Update")
+		data := broadcaster.SocketData{
+			MessageType: broadcaster.SecretUpdated,
+			Payload:     newObj,
+		}
+		endpoints.ResourcesChannel <- data
+	}
+}
+
+func secretDeleted(obj interface{}) {
+	logging.Log.Debug("Secret Controller Delete")
```

Comment:
Not a fan of using nouns for this, our usual debug messages are of the format (`in *method name` for example with any other useful params, here there's just a bunch of nouns)

Reply:
Will use the below comment suggestion for these.